src/admin/includes/formDefinitions/callbacks.php

```diff
@@ -0,0 +1,24 @@
+<?php
+function processInsert(){
+  $localvars = localvars::getInstance();
+  $formInfo  = $_POST['MYSQL'];
+
+  $roomID = isset($formInfo['roomID']) ? $formInfo['roomID'] : NULL;
+
+  $dupCheck  = reservationPermissions::duplicatePermissionsCheck($formInfo['resourceID'], $formInfo['resourceType'], $roomID, $formInfo['email']);
+
+  if (!$dupCheck) {
+   // return formInfo this is not a Duplicate Entry
+   return $formInfo;
+  } else {
+   $localvars->set("feedbackStatus",'<div class="error"> Error: Duplicate Entry </div><script>$(function(){$(".formErrors").hide();});</script>');
```

Comment:
You should be using the error handler. If there is a duplicate, and it is an error, it should be logged. You can then use the error handler to set the error message.

Using the error handler puts the error on EngineAPI's error stack, and allows printing of the error in a standard way.